Storage/src/StorageObject.php

```diff
@@ -715,9 +715,7 @@ public function downloadAsStream(array $options = [])
                     $endByte = $requestedBytes['endByte'];
 
                     // modify the range headers to fetch the remaining data
-                    $arguments[1]['headers']['Range'] = sprintf('bytes=%s-%s', $startByte, $endByte);
-
-                    return $arguments;
+                    $arguments[0] = $arguments[0]->withHeader('Range',  sprintf('bytes=%s-%s', $startByte, $endByte));
```

Comment:
I changed this because as we modify the request object in the other listener, we should be consistent here.

Reply:
I reverted this because it's not really required as part of this PR